src/glue/jobs/s3_to_json.py

```diff
@@ -16,6 +16,237 @@
+                "TimeOffsetSleepRespiration": (("OffsetInSeconds", int), ("Breaths", float))
+            },
+            "GarminStressDetailSummary": {
+                "TimeOffsetStressLevelValues": (("OffsetInSeconds", int), ("StressLevel", int)),
+                "TimeOffsetBodyBatteryValues": (("OffsetInSeconds", int), ("BodyBattery", int))
+            },
+            "GarminThirdPartyDailySummary": {
+                "TimeOffsetHeartRateSamples": (("OffsetInSeconds", int), ("HeartRate", int))
+            },
+            "GarminHealthSnapshotSummary": {
+                "Summaries.EpochSummaries": (("OffsetInSeconds", int), ("Value", float))
+            }
+    }
+    if dataset_identifier in garmin_transform_types:
+        this_data_type = garmin_transform_types[dataset_identifier]
+        for prop in this_data_type:
+            key_name = this_data_type[prop][0][0]
+            key_type = this_data_type[prop][0][1]
+            value_name = this_data_type[prop][1][0]
+            value_type = this_data_type[prop][1][1]
+            property_hierarchy = prop.split(".")
+            if len(property_hierarchy) == 1:
+                prop_name = property_hierarchy[0]
+                if prop_name in json_obj:
+                    array_of_obj = transform_object_to_array_of_objects(
+                            json_obj_to_replace=json_obj[prop_name],
+                            key_name=key_name,
+                            key_type=key_type,
+                            value_name=value_name,
+                            value_type=value_type
+                    )
+                    json_obj[prop_name] = array_of_obj
+            if len(property_hierarchy) == 2:
+                prop_name = property_hierarchy[0]
+                sub_prop_name = property_hierarchy[1]
+                if prop_name in json_obj:
+                    for obj in json_obj[prop_name]:
+                        if sub_prop_name in obj:
+                            array_of_obj = transform_object_to_array_of_objects(
+                                        json_obj_to_replace=obj[sub_prop_name],
+                                        key_name=key_name,
+                                        key_type=key_type,
+                                        value_name=value_name,
+                                        value_type=value_type
+                            )
+                            obj[sub_prop_name] = array_of_obj
+    return json_obj
+
+def get_output_filename(metadata: dict) -> str:
+    """
+    Get a formatted file name.
+
+    The format depends on which metadata fields we have available to us.
+    Metadata fields we can potentially use:
+        - type
+        - subtype
+        - start_date
+        - end_date
+
+    Args:
+        metadata (dict): Metadata derived from the file basename.
+
+    Return:
+        str: A formatted file name.
+    """
+    if metadata["type"] in DATA_TYPES_WITH_SUBTYPE:
```

Comment:
I know this was code that already existed but does datatypes with subtypes always have startdate?

Reply:
Yes, so far.